cfme/cloud/tenant.py

```diff
@@ -67,6 +67,18 @@ def wait_for_disappear(self, timeout=300):
         except TimedOutError:
             logger.error('Timed out waiting for tenant to disappear, continuing')
 
+    def wait_for(self, timeout=600):
+        return wait_for(self.exists, fail_condition=False, timeout=timeout,
+                        message='Wait for cloud tenant to appear', delay=10,
+                        fail_func=sel.refresh)
+
+    def update(self, updates, wait=True):
+        navigate_to(self, 'Edit')
+        updated_name = updates.get('name', self.name + '_edited')
+        create_tenant_form.fill({'name': updated_name})
+        sel.click(create_tenant_form.save_button)
+        # self.provider.refresh_provider_relationships()
```

Comment:
1) If it is required that the refresh take place for the update to show itself, it should be here as part of the update mechanism
2) I have already commented twice that the wait_for should take place inside the update method and should not be a separate method. For multiple reasons 
 a) You are only using it once, so there is no need to abstract it out
 b) The name is ambiguous
I don't even like the ```wait_for_disappear()``` that is used below. This really should just be a wait_for inside the method too, however since that is not really touched in this PR, I will ignore it.